lib/checkstl.cpp

```diff
@@ -3149,14 +3149,15 @@ void CheckStl::eraseEndIterator()
             if (args.size() != 1) // empty range is ok
                 continue;
 
-            if (const ValueFlow::Value* itValue = args[0]->getKnownValue(ValueFlow::Value::ValueType::ITERATOR_END)) {
-                if (itValue->intvalue == 0)
+            if (const ValueFlow::Value* endVal = args[0]->getKnownValue(ValueFlow::Value::ValueType::ITERATOR_END)) {
```

Comment:
This should probably be a for loop as we should warn for possible values as well:

```cpp
if (it == v.end()) {}
v.erase(it);
```